msrestazure/azure_active_directory.py

```diff
@@ -647,15 +667,16 @@ def __init__(self, port=50342, **kwargs):
         super(MSIAuthentication, self).__init__(None)
 
         self.port = port
+        self.msi_conf = {k:v for k,v in kwargs.items() if k in ["client_id", "object_id", "msi_res_id"]}
```

Comment:
still debating a bit, but to get interface clear, we should throw when user assigned identity info is supplied for a  webapp

Reply:
Let's do that for now, but I won't be surprised if this comes to WebApp at some point...

Reply:
Done (see last commit)